model/project.go

```diff
@@ -2296,3 +2297,61 @@ func GetVariantsAndTasksFromPatchProject(ctx context.Context, settings *evergree
 	}
 	return &variantsAndTasksFromProject, nil
 }
+
+// ReadOutputPath reads the content of a file and parses it into a github.CheckRunOutput struct
+// if it fails validation, it returns an error
+func ReadAndValidateOutputPath(file string) (*github.CheckRunOutput, error) {
+	checkRunOutput := &github.CheckRunOutput{}
+
+	if file == "" {
+		return checkRunOutput, nil
+	}
+
+	if err := utility.ReadJSONFile(file, &checkRunOutput); err != nil {
+		return nil, err
+	}
+
+	return checkRunOutput, validateCheckRuns(checkRunOutput)
+}
+
+func validateCheckRuns(checkRun *github.CheckRunOutput) error {
+	if checkRun == nil {
```

Comment:
Because it would hit this case when someone specified a path, but it didn't parse to a checkrun object. For example, if they did something like: 

{
   output: <the output>
}

instead of just 

{
   <the output>
}

So it would be an error case because they attempted to create it but didn't